core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -75,8 +75,40 @@ impl Cast {
     fn cast_array(&self, array: ArrayRef) -> DataFusionResult<ArrayRef> {
         let array = array_with_timezone(array, self.timezone.clone(), Some(&self.data_type));
         let from_type = array.data_type();
-        let cast_result = cast_with_options(&array, &self.data_type, &CAST_OPTIONS)?;
-        Ok(spark_cast(cast_result, from_type, &self.data_type))
+        let to_type = &self.data_type;
+        let cast_result = match (from_type, to_type) {
+            (DataType::Utf8, DataType::Boolean) => Self::spark_cast_utf8_to_boolean::<i32>(&array),
+            (DataType::LargeUtf8, DataType::Boolean) => {
+                Self::spark_cast_utf8_to_boolean::<i64>(&array)
+            }
```

Comment:
We probably need to handle dictionary types too. Maybe as a follow up.

Reply:
Is this for casting from `Dictionary.value_type` to `Boolean`?

Reply:
Yea

Reply:
i.e., `Dictionary(_, DataType::Utf8)` and `Dictionary(_, DataType:: LargeUtf8)`

Reply:
Sure, please feel free to create an issue so i can have a look after this PR. Thanks